api/controllers/ApplicationTypeController.js

```diff
@@ -54,93 +55,87 @@ module.exports = {
-                            error_report: false,
-                            changing: false,
-                            form_values: false,
-                            submit_status: req.session.appSubmittedStatus,
-                            current_uri: req.originalUrl,
-                            user_data: userData,
-                            back_link: req.session.startBackLink,
-                            //disableStandardServiceSection: disableStandardServiceSection
-                        });
-                    });
-                });
-            });
-
-        }
-
         req.session.appId = false; // reset the appId so a new session is used
         // set initial submit status to false, meaning it application has not yet been submitted
         req.session.appSubmittedStatus = false;
 
-        return res.view('applicationForms/applicationType.ejs', {
+        const userModels = getUserModels(req._sails.config.userServiceSequelize);
+
+        return this._renderServiceSelectionPage(req, res, userModels);
+    },
+
+    _renderServiceSelectionPage(req, res, userModels) {
+        const userLoggedIn = HelperService.LoggedInStatus(req);
+        const userData = HelperService.getUserData(req, res);
+        const pageData = {
             application_id: 0,
             userServiceURL: sails.config.customURLs.userServiceURL,
-            error_report: false,
+            error_report: Boolean(req.query.error),
             changing: false,
             form_values: false,
             submit_status: req.session.appSubmittedStatus,
             current_uri: req.originalUrl,
-            user_data: HelperService.getUserData(req,res),
-            back_link: req.session.startBackLink,
-            //disableStandardServiceSection: disableStandardServiceSection
+            user_data: userData,
+        }
+
+        if(userLoggedIn) {
+            return this._addUserAccountToSession({req, res, userModels, pageData});
+        }
+
+        return res.view('applicationForms/applicationType.ejs', {
+            ...pageData,
+            back_link: '/',
         });
+    },
 
+    _addUserAccountToSession({req, res, userModels, pageData}) {
+        return userModels.User.findOne({where: {email: req.session.email}}).then((user) => {
+            return userModels.AccountDetails.findOne({where: {user_id: user.id}}).then((account) => {
+                const standardAppCountQuery = 'SELECT count(*) FROM "Application" WHERE "user_id" =:userId and "serviceType" = 1 and "createdAt" > NOW() - INTERVAL \'' + sails.config.standardServiceRestrictions.appSubmissionTimeFrameInDays + ' days\' and ("submitted" =:submitted OR "submitted" =:queued)';
+
+                console.log('this gets hit 3', account)
+                return sequelize.query(
+                    standardAppCountQuery,
+                    { replacements: {userId: user.id, submitted: 'submitted', queued: 'queued'},
```

Comment:
submitted: true? are there other states?

Reply:
Yep there's 'queued', 'draft' and false. Would make sense to have this as a frozen object/enum somewhere actually so that other's don't get confused.

Happy to create a follow up refactor ticket to look into this in the future.